### PR TITLE
Add 2016 api-version and tests

### PR DESCRIPTION
**Pull request by vkc12uec**
Support new Api version in our ARM resource creation.

contact:
vivekcha@microsoft.com

### CODE REVIEW

File: schemas/2016-07-01-preview/Microsoft.ServerManagement.json [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+          ]
+        },
+        "type": {
+          "enum": [
+            "Microsoft.ServerManagement/gateways"
+          ]
+        },
+        "properties": {
+          "type": "object",
+          "upgradeMode": {
```
Comment (TianoMS) [inline schemas/2016-07-01-preview/Microsoft.ServerManagement.json]: should be put in an "properties" definition. like this:
"properties": {
    "upgradeMode": ...
}

**Comment by TianoMS** [comment]
@vkc12uec Please address the comments.

**Comment by vkc12uec** [comment]
I was OOF. Will fix them now.

**Comment by vkc12uec** [comment]
@TianoMS  Hi Tiano, I have now used autoRest tool to generate the ARM schema. I have push changes to my repo. Looks like the PR has updated. 

Can you take a look ? Thanks!

File: schemas/2016-07-01-preview/Microsoft.ServerManagement.json [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+        }
+      },
+      "required": [
+        "type",
+        "apiVersion",
+        "properties"
+      ],
+      "description": "Microsoft.ServerManagement/nodes/sessions"
+    },
+    "nodes_sessions_features_pssessions": {
```
Comment (TianoMS) [inline schemas/2016-07-01-preview/Microsoft.ServerManagement.json]: I don't see this referenced in deploymentTemplate.json. Do we need this definition?

File: schemas/2016-07-01-preview/Microsoft.ServerManagement.json [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+    "SessionParameters_properties": {
+      "type": "object",
+      "properties": {
+        "userName": {
+          "type": "string",
+          "description": "User name to be used to connect to node"
+        },
+        "password": {
+          "type": "string",
+          "format": "password",
```
Comment (TianoMS) [inline schemas/2016-07-01-preview/Microsoft.ServerManagement.json]: "password" is not a built-in format.

**Pull request merged**